src/utils/cosmosdb_model_versioned.ts

```diff
@@ -166,6 +172,66 @@ export abstract class CosmosdbModelVersioned<
     );
   }
 
+  /**
+   * Add the given ttl at all the version of the document identified by
+   * the searchKey
+   */
+
+  public updateTTLForAllVersion(
+    searchKey: DocumentSearchKey<TR, ModelIdKey, PartitionKey>,
+    ttl: NonNegativeNumber
+  ): TE.TaskEither<CosmosErrors, ReadonlyArray<TR>> {
+    const partitionKey = searchKey.length === 1 ? searchKey[0] : searchKey[1];
+    return pipe(
+      this.findAllVersionsByPartitionKey(searchKey),
```

Comment:
maybe we don't need to get **ALL** the documents versions, what if we take only the last (cheaper query) and iterate the update on all the documents version from 1 to the last.

Reply:
Very cleaver! We then can use `generateVersionedModelId` to generate each document id from 0 to last version.